tests/lib/rules/no-unused-prop-types.js

```diff
@@ -1382,6 +1382,20 @@ ruleTester.run('no-unused-prop-types', rule, {
       ].join('\n'),
       parser: 'babel-eslint',
       parserOptions: parserOptions
+    }, {
+      // Destructured state props in `componentDidUpdate` [Issue #825]
+      code: [
+        'class Hello extends Component {',
+        '  static propTypes = {',
+        '    something: PropTypes.bool',
+        '  }',
+        '  componentDidUpdate ({something}, {state1, state2}) {',
+        '    return something;',
+        '  }',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint',
```

Comment:
Does the default parser support object destructuring? I can't seem to get it to work. I keep getting: `Parsing error: Unexpected token }`